source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
    :language: ruby
    :start-after: # start-embedded-destroy-all
    :end-before: # end-embedded-destroy-all
+
+Association Behaviors
+---------------------
+
+When working with associations in {+odm+}, there are various behaviors that
+allow you to customize how associations are handled in your application. The following
+sections describe common association behaviors.
+
+Extensions
+~~~~~~~~~~
+
+Extensions allow you to add specific functionality to an association. You can
+define an extension on an association by specifying a block in the association
+definition, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-extensions
+   :end-before: # end-extensions
+   :emphasize-lines: 4-6
+
+Custom Association Names
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``class_name`` macro to specify a custom class name for an
+association. This is useful when you want to use a different name for an
+association than the name of the class. The following example uses the
+``class_name`` macro to specify that an embedded association called ``records``
+represents the ``Album`` class:
```

Comment:
q: if you hadn't specified this, what would the association be called? i'm not very familiar with mongoid or associations

Reply:
If it's not specified it just uses the name of the class - so in this example if you just said `embeds_many :records` it would use the Record class (mongoid automatically translates plural -> singular)